### PR TITLE
Fixes #1389

### LINKED ISSUE #1389: Create a DP for has-soma-location pattern
So far we have a DP documented for cells part of anatomy:

https://obophenotype.github.io/cell-ontology/patterns/cellPartOfAnatomicalEntity/

But none for has-soma-location(HSL)

We need to add these, and make the ontology consistent (which @shawntanzk has been doing -see #1337) 

We need to document whether

 1. We *only* use HSL for neurons->uberon
 2. We use a mixture of part-of and HSL as appropriate
     - typically use HSL for equivalence axioms (principle: weaker properties in EC -> stronger inferences)
     - add supplemental part-of hidden GCIs if we know the neuron is not projecting outside the anatomical area

While 1 is both safer and easier to implement, I worry we will lose crucial inferences in CL

In particular I worry about loss in:

 - gene expression queries that propagate over part-of
 - entailments in ontologies that use CL for logical definitions

To restore these inferences we will need to add a lot of hidden GCIs

## Alternative proposal: radical mereological inclusivity

We abandon HSL, we go back to using simple powerful part-of relations.

We conceive of uberon brain regions such as hippocampus as having 'classic boundaries' at one level of granularity, but when zoomed in to the subcellular level the hippocampus itself has synaptic parts that interleave with other brain regions. These could be described using simple projects-into relationships.

This would require some minor changes in uberon: the existing adjacency relations (not that useful IMO) would need to be changed to something like 'grossly adjacent to'.

Gene expression queries would be more inclusive but arguably more correct since expression is in the soma.

### PR DESCRIPTION
**Pull request by dosumis**
Fixes #1389

- #1389

### CODE REVIEW

File: src/patterns/dosdp-patterns/neuronHasSomaLoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  anatomical_entity: "'anatomical entity'"
+
+data_list_vars:
+  pubs: "'xref'"
+  # Use this for recording reference using PMID/DOIS
+
+name:
+  text: "%s of %s"
+  vars:
+    - cell
```
Comment (shawntanzk) [inline src/patterns/dosdp-patterns/neuronHasSomaLoc:35]: Should this var be 'neuron'?

Comment (dosumis) [reply src/patterns/dosdp-patterns/neuronHasSomaLoc:35]: Yes.  Please fix.  Thanks!

**Comment by bvarner-ebi** [comment]
Checking in on this PR before upcoming CL release.
No description, no action for a month and referenced ticket is already closed.
Can this PR be merged or closed?

**Comment by anitacaron** [comment]
The change in this PR is not relevant to the CL release.

File: src/patterns/dosdp-patterns/neuronHasSomaLoc (hunk 1 above)
Comment (anitacaron) [reply src/patterns/dosdp-patterns/neuronHasSomaLoc:35]: All other 'cell' should be changed to 'neuron'?

Comment (anitacaron) [reply src/patterns/dosdp-patterns/neuronHasSomaLoc]: done

**Review by anitacaron — APPROVED** [review]

**Comment by bvarner-ebi** [comment]
@dosumis, can this be merged and closed?

**Pull request merged**